internal/sqladapter/testing/adapter.go.tpl

```diff
@@ -1131,14 +1130,15 @@ func TestBatchInsertReturningKeys(t *testing.T) {
 
 	go func() {
 		for i := 0; i < totalItems; i++ {
-			batch.Values(fmt.Sprintf("artist-%d", i))
+			batch.Values <- fmt.Sprintf("artist-%d", i)
 		}
-		batch.Done()
+		close(batch.Values)
 	}()
 
-	for q := range batch.Next() {
-		var keyMap []struct{ID int `db:"id"`}
-		err := q.Iterator().All(&keyMap)
+	var keyMap []struct {
+		ID int `db:"id"`
+	}
+	for batch.Next(&keyMap) {
```

Comment:
This for without a range looks good, the problem is that not every `INSERT` query will return keys. If the user didn't provide a `Returning()` method this won't work.

Also, using `RETURNING` turns the `INSERT` query into something that returns rows, but the usual way to execute `INSERT` is by doing `Exec()`, which doesn't return any rows. Basically `Exec()` must be used when no `Returning()` parameter was specified.